index.js

```diff
@@ -0,0 +1,274 @@
+/**
+ * Full fhirconvert function: validates, converts, then identifies variables.
+ * @returns converted fhirpath expression
+ */
+export function fhirconvert(str, vars, funs) {
+    if (validate(str, vars, funs)){
+        return varfind(convert(str), vars);
+    }
+    else {
+        return null;
+    }
+}
+
+/**
+ * Verifies normal syntax by confirming var names, function names,
+* and number of parenthesis.
+* @param {string} str - inputted normal syntax expression
+* @returns boolean, valid or invalid
+*/
+export function validate(str, vars, funs) {
+var lcount = 0;
+var rcount = 0;
+var len = str.length;
+var func = "";
+for (var i=0; i<len; i++) {
+    if (str[i] == "(") {
+        lcount += 1;
+    }
+    if (str[i] == ")") {
+        rcount += 1;
+    }
+
+    if (((/[a-z]/).test(str[i]))) {
+        if (!(/[a-zA-Z]/).test(str[i+1]) || !(/[a-zA-Z]/).test(str[i-1])) {
+            if(!vars.includes(str[i])) {
+                return false;
+            }
+        }
+    }
+    if ((/[A-Z]/).test(str[i])) {
+        func = func + str[i];
+    }
+    else {
+        if (funs.includes(func) || func == "") {
+            func = "";
+        }
+        else {
+            return false;
+        }
+    }
+}
+return (lcount == rcount);
+}
+    
+/**
+ * Identifies convertable functions in expression and converts them recursively.
+ * @param {string} str - inputted normal syntax expression
+ * @returns expression with converted functions
+ */
+export function convert(str) {
+    var count = 0;
+    if (str.includes("^")) {
+        var i = str.indexOf("^");
+        var base = lfind(str, i);
+        var power = rfind(str, i);
+        str = str.slice(0, i-base.length) + "(" + base + ".power(" + power + ")" + ")" + str.slice(i+power.length+1);
+        count += 1;
+    }
+    if (str.includes("CEILING")) {
```

Comment:
Since most of these functions are handled just the same, perhaps you could use a loop on the function name (at least for most of them)?

Reply:
Just incorporated it.